File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.cc
#include <iostream>
#include <list>
#include <cmath>
#include <algorithm>

using namespace std;

int main() 
{
    while(true)
    {
        string str;
        string reverse_str;
        cin >> str;
        reverse_str = str;
        reverse(reverse_str.begin(), reverse_str.end());

        if(str == "0")
        {
            break;
        }
        int num;
        num = ceil(str.length() / 2);

        string result = "yes";
        for(int i=0; i<num; i++)
        {
            int temp = i + 1;
            // cout << "str[i] : " << str[i] << " str[-temp] : " << reverse_str[i] << endl;
            if(str[i] == reverse_str[i])
            {
                result = "yes";
            }
            else
            {
                result = "no";
                break;
            }
            
        }
        cout << result << endl;
    }
}

File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.py
import math

while True:
    N = input()
    string_list = list(N)
    
    if '0' == N:
        break
  
    num = math.ceil(len(string_list) / 2)
    
    result = "yes"
    for i in range(num):
        if string_list[i] == string_list[-(i+1)]:
            result = "yes"
        else:
            
            result = "no"
            break
    print(result)
